sklearn/preprocessing/tests/test_label.py

```diff
@@ -536,3 +541,33 @@ def test_inverse_binarize_multiclass():
                                                    [0, 0, 0]]),
                                        np.arange(3))
     assert_array_equal(got, np.array([1, 1, 0]))
+
+
+@pytest.mark.parametrize('engine', ['numpy', 'python', 'pandas'])
+@pytest.mark.parametrize(
+        "values, expected",
+        [(np.array([2, 1, 3, 1, 3], dtype='int64'),
+          np.array([1, 2, 3], dtype='int64')),
+         (np.array(['b', 'a', 'c', 'a', 'c'], dtype=object),
+          np.array(['a', 'b', 'c'], dtype=object)),
+         (np.array(['b', 'a', 'c', 'a', 'c']),
+          np.array(['a', 'b', 'c']))],
+        ids=['int64', 'object', 'str'])
+def test_factorize_encode_utils(engine, values, expected):
+    # test that all different encoders are equivalent
+
+    if engine == 'numpy':
```

Comment:
you can parametrize this part as well and mark one of the solution as skipif with pytest, isn't it?